### ISSUE TITLE
iPhone video starts with a black screen

### ISSUE THREAD

**Issue by BPerlakiH** [kiwix/kiwix-apple#916]
After pressing play, the video starts in full screen, but the screen is black.
If we close full screen and go fullscreen again or toggle it via picture in picture it works fine. These steps need to be repeated for each video, even if they are on the same “tab”.

Content used (currently in the "New" section):
- [TED collective (31 Jul 2024)](https://download.kiwix.org/zim/ted/ted_mul_collective_2024-07.zim.meta4)
- [TED humanities (02 Aug 2024)](https://download.kiwix.org/zim/ted/ted_mul_humanities_2024-08.zim.meta4)

**Comment by kelson42** [comment]
I can not reproduce this bug with iPadOS on iOS18. What are the exact reproduction steps?

**Comment by BPerlakiH** [comment]
It is iOS 17 specific and only on the iPhone.

**Comment by kelson42** [comment]
@BPerlakiH So, this is working on iPhone with iOS18?

**Comment by BPerlakiH** [comment]
I am not sure if it's working on iOS18, maybe it's a bit too early for that, as I do not have iOS 18 installed yet, and currently it's not yet released.
I will have a look via Beta (Xcode 16 Beta, and iOS 18 Beta), but that is only on the simulator, which might not be accurate.

**Comment by kelson42** [comment]
@BPerlakiH Please do, testing with iPadOS 17 as well.

**Comment by kelson42** [comment]
Please share here the table of error for iPadOS/iOS and iOS17 and 18.. The exact step-by step procedure + screencast are needed.

**Comment by kelson42** [comment]
@BPerlakiH Split also in teo different issues the autofullscreen issue and the black box issue.

**Comment by BPerlakiH** [comment]
@kelson42  I have updated my PR, that fixes the black screen issue for all (known to me) content. If we are good to go with that solution, the auto-full screen won't be an issue any more, as it seems to be only content related: Depending only on the content loaded it either goes full screen directly or not, so I am not sure if that needs to be "fixed".

Will have a look on iOS 18 as well...

**Comment by BPerlakiH** [comment]
Well... iOS 18 is still in beta, so many things are not working.
Eg. this code is crashing:
``
webView.setValue(view.safeAreaInsets, forKey: "_obscuredInsets")
``
but even if I take that out, the webm videos are not playing, just spinning and loading:
<img width="850" alt="Screenshot 2024-08-15 at 19 27 00" src="https://github.com/user-attachments/assets/adc1d6ff-dd0b-4dae-a160-c56d6a53fd4d">
<img width="478" alt="Screenshot 2024-08-15 at 19 19 15" src="https://github.com/user-attachments/assets/5f7d2c63-f8d9-4f8f-b0f0-7910d901539e">

So it's not helpful at all, once we will have a Golden Master release from Apple, it's worth to re-test.

**Comment by BPerlakiH** [comment]
TLDR:
this only affects iOS 17 no the iPhone.
iPad on iOS 17 is OK.

The PR is fixing this issue specifically for iPhone on iOS 17.
iOS 18 is still in beta, and unreliable to compare with.

**Comment by kelson42** [comment]
This is a very serious bug and considering we have been working since May on these topics, it sounds to me we might face a regression here. https://github.com/openzim/ted/pull/185 should have fixed it.

**Comment by rgaudin** [comment]
Thank you ; I confirm that as far as I can tell, this is an upstream issue in **WKWebView+iOS 17+WebM+iPhone** (I only tested on iPhone so I trust you gays on the last condition).

Reprocase is very simple as the following doesn't work. Remove the [poster](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/video#poster) attribute and it works…

```html
<html>
<body>
<video poster="thumbnail.png" src="video.webm" />
</body>
</html>
```

- [stackoverflow question](https://stackoverflow.com/questions/78255334/ios-17-video-shows-a-black-screen-when-opening-in-full-screen-mode-with-a-post) about it.

So let's go with the workaround for now.

**Comment by BPerlakiH** [comment]
Opened a separate issue for the "auto" full screen issue here: https://github.com/kiwix/kiwix-apple/issues/933
